### DISCUSSION TITLE
Response body as IImageProvider?

### DISCUSSION THREAD

**Discussion by robbaman** [SixLabors/ImageSharp.Web#134] [Q&A]
Hi there! 

I was wondering, is there a way to use the response body as an image provider of sorts? We have a dynamic system of file providers that yield files from api controllers (including fairly often images) that could really benefit from ImageSharp.Web post-processing in many cases.

The thing is, we pretty accurately detect that the call is meant to be processed by Uri's used, but I need the input for the ImageSharp.Web middleware to be the Response.Body that is generated by the ApiController. 

Is this a thing that can be done by implementing some interfaces or applying some specific configuration? I've looked at the source, but it's pretty complex, so if you could point me into the direction I should be looking at, it would be greatly appreciated.

Thanks in advance.

Regards,

Robert

**Comment by JimBobSquarePants** [comment]
Theoretically yes you could... 

 `IImageProvider` accepts `HttpContext` for all the method parameters so you could read/check/resolve the response body from there.  
 
 You'd have to ensure it is rewound after reading so that the output is overwritten.
